### DISCUSSION TITLE
Firmware not uploading. [SOLVED]

### DISCUSSION THREAD

**Discussion by NightShift00** [roleoroleo/yi-hack-MStar#484] [General]
I've got a YI Home Model Number: 9FUSY31SF5. The 1st party firmware currently on the camera is 12.2.0.2_202212191129. I've tried flashing multiple custom firmware versions onto the camera using a 2Gb FAT32 formatted micro SD and the procedure dictated in the instructions and every time the camera just ignores it and starts up like normal. I never get the flashing yellow light. Can anyone help?

Edit 1: Okay, I've realized the issue is my official firmware is brand new and the current hacks are based on older builds. I'm currently trying to roll back the firmware to v8, but the camera steadfastly refuses to flash any firmware at all off the SD. 

Edit 2: Okay, I got. NVM. I was finally able to rolled back the firmware to the official YI v8 build. The camera wasn't seeing the card, but I was able to fix this by doing a clean format in windows, putting the blank card in the camera, letting the camera do it's thing to it, putting it back in my PC, ignoring windows' requests to scan and repair the card, putting the firmware on the root path and reinserting the camera into the camera. Here's where to get older official YI firmware.
https://www.kamihome.com/firmware/ 
I will say though I never saw the described blinking yellow light. I simply got a solid yellow light, followed by a brief flash of blue. The YI hack firmware is working now though.

**Comment by roleoroleo** [comment]
What cam are you using?
Probably it's not MStar based.

**Reply by NightShift00** [reply]
The box is calling it "YI 1080p Home Camera"  and there's a sticker applied to the box and camera after the fact that says AI+. UPC: 6970171172421.

**Reply by roleoroleo** [reply]
What about firmware version?
I mean YI firmware v8.

**Reply by NightShift00** [reply]
I never made note of the out of box firmware before it updated to the newest. But the rollback firmware that worked was 8.2.0.0A_202012041528.

**Reply by roleoroleo** [reply]
Your cam is Allwinner based, not MStar:
https://github.com/roleoroleo/yi-hack-Allwinner

**Reply by NightShift00** [reply]
Would that explain the indicator light behavior I saw?

**Comment by roleoroleo** [comment]
I don't remember the exact sequence of the indicator for Allwinner based cam.
But is very different from MStar.
And your have to download the hack from this repo:
https://github.com/roleoroleo/yi-hack-Allwinner
suffix y20ga.
Your cam will not load MStar based hack.
